src/app/app-layout/app-header.component.ts

```diff
@@ -114,6 +119,7 @@ export class AppHeaderComponent implements OnInit {
       )
       .subscribe((customData: any) => {
         this.header.select(customData);
+        this.cdr.detectChanges();
```

Comment:
The component uses OnPush change detection strategy, which is great for optimization, but can lead to hard to find bugs. In this case, the header's data is updated in such a way that a change detection does not trigger with OnPush strategy, so we need to manually run it. This should also resolve the issue where on refresh, the Home link wasn't being highlighted